### DISCUSSION TITLE
A survey is needed to see if good defaults are being used

### DISCUSSION THREAD

**Discussion by Gene703122** [nvaccess/nvda#13427] [Ideas]
I have long been dissatisfied with the defaults concerning what is read in browse mode.  This would mostly affect Internet use, though it is used elsewhere.  

On what basis are defaults chosen?  Shouldn't users be consulted about some of them to see the satisfaction or lack of satisfaction with certain questionable defaults?

It appears that the more NVDA can announce, the more it does announce by default.  In a recent version, "figure" and "out of figure" started being announced.  This is meaningless information unless you are editing or designing a page.  It is clutter and does me no good to hear figure when I move to a certain link to a story on The New York Times page.  "Out of figure" tells me nothing useful either.

In the last version or two of NVDA, all sorts of shapes like right pointing triangle, hollow left pointing triangle, hollow down pointing small triangle, and filled left pointing small triangle, are spoken when encountered and there are perhaps fifty or more other shapes including various circular shapes.

Why are these all or almost all set to none as the punctuation level at which they are spoken.  They should be set to all or at least to most.

There is no setting to turn all these shape announcements off other than to use one that turns all useful announcements off such as smiling face.  This isn't reasonable nor does it make sense.  

At times, right pointing triangles serve in place of bullets.  Its distracting and maddeningly annoying to be reading an article and hear a number of points being enumerated about something and hear right pointing triangle before every one.
It tells me nothing useful as a general reader.  If I'm proofing an article, I'd want to know they are there.  As a reader. I absolutely do not.

And why is bullet set to some and most or perhaps all other bullets such as triangular bullets set to none?  
As a reader, I think none should mean none.  It should be where you don't hear such things as bullets or shapes.

Why are bloc quotes announced by default?  Do most people want lists to be announced?  

A lot of people just accept technology as it is.  That doesn't mean they like everything a program does, but they don't complain.  They may be intimidated and believe there are reasons they don't know and shouldn't contest behaviors.   They may know they don't like something but not know how to change it.  I really wonder how many NVDA users know how to work with dialogs such as document formatting or know they exist, much less what they do.  

Defaults should be what is best for the most people and should also be at times, what is Goode for beginners.  and inexperienced users.  I may discuss an instance of that in another discussion topic.

I've never heard of even one survey attempting to find out what users want.  I don't know how defaults are decided on but if developers decide on them without meaningful input from a wide array of users, defaults reflect what developers think should be done which can't be assumed to be what users want.  A lot of defaults are sound.  But a small number of unsound ones can cause a lot of people a lot of annoyance and in some cases meaningful inconvenience.  

Perhaps, along with the welcome dialog, a survey should come up to encourage users to take it and a submission mechanism such as a button would be provided.  I think NVDA isn't taking effective steps to encourage a lot of users to give feedback.

It may be that there aren't enough volunteers to have such a feature.  Someone would have to read and compile results.  Perhaps volunteers might be recruited for that purpose.  I don't have the technical knowledge to work on add-ons or development.  I might volunteer for working with surveys.  

Rightly or wrongly, a lot of people are intimidated by Github and a lot of users don't even know about it.

**Comment by XLTechie** [comment]
I haven't been a "new user" for a very long time, and even when I was, I am someone who likes details so am not bothered by a lot of what Gene targets here.

That said, I do find his over all argument to be resonant. I have thought that NVDA does have a tendency to be both too verbose about web entities, and inconsistent about some of its defaults.

I haven't noticed all of the triangles and such that Gene points out, although I have noticed things like "double right pointing white triangle" and "black circle" being used as bullet substitutes. While I found such to be overly verbose, short of replacing them with "bullet", I don't know what's to be done. I certainly don't want to avoid knowing that what I'm hearing are list items.

Because contrary to Gene, I do like knowing that I'm in a list. For three reasons (hereby placed in a list, because I can):
* it lets me know that I have navigation options (such as comma, to quickly get out of the list if I don't want to hear it--often the case with menus of links);
* so that if I want to find previous list items to inspect in greater detail, I know I can just hit i/I to move around and investigate them; and
* if I get further down in the document, and realize something in the list is info that I need to hear again, I can quickly press L to jump back to the list, and hear it. The same with tables, for that matter.

I'm one of those people who rarely uses say all on web pages. There's usually too much extraneous content on the web, and I value my time too much to waste it hearing groups of links, in-line ads, sidebars, and so on. I prefer to jump right to, and through, valuable content.
I have "automatic say all on page load" set, but only to hear that the page has loaded. I never actually let it read from the top.

As to blockquotes: I most definitely want to hear those, maybe even more than lists (in say all, anyway). How else am I to know whether what I am reading is original content from the author, or borrowed content from some other source? That is not always clear from context, and sometimes it makes all the difference to how one interprets what is being read.

These things said, I do not consider myself "the average user". I'm a detail oriented person in general, and like as much as possible to understand the layout of what I'm reading. I tend (on automatic) to picture the document layout in my head as I'm reading it. This might come from being able to see some of those details more as a child.
Further, I'm a programmer. I have a built in mental filter for hearing undesired punctuation and presentational details, and being able to ignore them/not be bothered by them, if they aren't relevant at the moment.
This, in part, has made me a very good proof reader/copy editor.

**I do think Gene has made a good case for an issue to reclassify alternative bullets as bullets, and then to obey the current punctuation level's take on how (not) to indicate them.**
Perhaps an option somewhere for "list indicators", as a combo/radio button: speak original, speak as bullets, honor bullets punctuation level, don't speak/braille.

**Comment by XLTechie** [comment]
The survey idea is interesting, but I am curious how such a thing could be structured.

A long list of examples, with "would you rather hear/not hear/hear something else" for each, would be tedious and so situationally subjective as not to be informative.

Further, I think it is reasonable to suspect that new users often don't know what they want. If they are very new, the answers they give on a survey about what they *think* they might want, and the actuality of what they prefer in practice, are likely to be very different things.
And by the time they become "old users", they have likely customized their reading environment to be so far from the defaults, that asking what defaults they would have preferred as new users would be meaningless.

This gets to the core, I think, of why things are as they are.

You point out, Gene, that users may not know how to work with document preferences, etc., and may be intimidated or afraid to mess with things there. I agree. But for that very reason, things like figure announcement, list announcement, table announcement, and on and on and on, are active by default.

Is it not better for the user to hear too much while desiring to hear less, and eventually learn or be inspired to figure out *how* to hear less?
The alternative to that is to decide for the user what s/he is likely to want to hear, and, in the case of some (many?) users, the knowledge that there is even more that can be heard, may never arrive.

Figures, lists and blockquotes annoy you, so you figure out how to turn them off once a certain threshold of botheryness is reached. But if you don't know they are there to begin with, and you assume in good faith that the screen reader is telling you everything it reasonably can about the page you are reading, you may never seek to turn them on because you don't know what you're missing.

This is a careful balancing act, and personally I think I would put more weight on the side of too much information about formatting being preferable to not enough.

I would, however, agree that the *way* NVDA presents some of this information could be improved. Is it really necessary to say "out of link"? "out of link out of figure out of heading level 3", etc.? This is why I am a major proponent of optional earcons, as iOS increasingly allows, to cut down on the verbal clutter.

Perhaps there are more concise ways of notifying the user that an element, or worse a group of elements, is no longer focused, then just verbally unwinding the list of containing elements with an "out of" before each.
If so, I'm not sure what it is, but personally I would be very interested in that angle.

**Reply by Gene703122** [reply]
This is a long response.  Because I've written long messages, I won't write more for a time and I'll see how the discussion goes.  

Almost by definition, anyone participating in this discussion isn't a typical user.  Nor, almost by definition, are developers.  

I am not assuming that developers don't know what typical users want, I haven't discussed that with developers and I don't know that, nor do I have any examples of how defaults are arrived at.  But neither do I have any evidence that they really know what people want.  As I said, most defaults are sound but a small number of bad defaults can result in a lot of annoyance and at times, in unintended consequences.  How are defaults determined?

At times, ideology replaces common sense.  A system tray dialog is not offered in the core, yet the manual, last I checked awhile ago, offered no meaningful and detailed discussion of how to work with the system tray directly.  When I asked about this on the large NVDA users' list, a developer told me that explaining this is not explaining how NVDA works and is therefore not discussed in detail.  I'll check to see if this has been changed but it was as I described for years.  

If you are not including something that has been expected in Screen-readers since Windows 98 and you don't instruct people in the users' Guide how to use the actual interface, you are letting developer ideology replace common sense.  

To return to automatically save settings, I strongly believe that from the standpoint of a student and a teacher, save settings automatically on exit is a bad default.  One of the best teachers in the field today commented in a class I took many years ago that Window-eyes, which was still around then,  had an advantage over JAWS because settings weren't saved automatically.  That allowed students to try things and not permanently change things.  

Also, this default allows settings to be accidentally saved or mistakenly saved.  I discussed this in a ticket but I'm bringing it up here because my question is how developers decided on this default.  Were trainers with good reputations and experience consulted?  Were cases such as the following considered:
Someone uses NVDA control shift r three times to revert to factory defaults to change something.  While in this state, the computer spontaneously reboots.  All the settings the user made previously are lost.  The factory configuration is saved when the computer and NVDA shut down.  This is a flaw that should not exist.  

Say I want to hear a document using unusual settings like punctuation set to all and attributes announced.  If I change these settings intended for one session and my computer spontaneously reboots or I forget and switch to another screen-reader, all these changes are saved and have to be undone.  Were any of these sorts of situations considered?

As far as knowing if you are in a list, I have list notifications off but doesn't NVDA announce by default if you are in one and when you leave one?  

If I'm reading an article, I don't want to hear, The President will make the following points in his address:
list with five items, and then hear the points, then hear out of list.  
Without some way of trying to determine any sort of at least possible majority opinion, how do developers know what people want in this respect?  

You talked about it being better to hear too much instead of too little.  Then why isn't punctuation set to all or most?  I'm being a bit facetious but I'm pointing out a flaw in your logic.  Why is this alright in some cases and not in others and what purpose is served?  

You, Luke, appear to believe that users in general learn how to change things they don't like over time.  You appear to believe that a lot of users, perhaps the majority, learn a good deal about NVDA.  Unless you have evidence to back that up that is somehow systematic and reliable, that is an opinion.  I believe that a lot of people, I don't know if it is a majority, don't learn a lot about NVDA or programs.  I don't have systematic evidence but I base my opinion on questions I see asked on various lists and practices i have observed for years.  

Look at the unending terror of ribbons.  Look at how many people ask questions like how do you turn this on or off in NVDA when if they looked through menus and/or dialogs or documentation, they could easily find out.  I'm not expressing either approval or criticism of this lack of learning or of asking such questions.  I'm stating some reasons why I believe that a lot of users, I don't know if it is the majority, don't learn nearly as much over time as you may think.  But again, without some way of getting information, we are discussing opinions.  

A user survey, I believe, might be very useful.  And perhaps asking trainers for comments about what they like and don't like about NVDA in training might be useful. That might be an entirely separate outreach.

the survey could ask the participants level, new user, moderately knowledgeable user, experienced user, to give three examples.

You wouldn't ask question after question about defaults or what people want to hear.  You could have a question such as what particularly annoys you about NVDA?  You should have a few questions about whether hearing this or that announcement such as list, bloc quote, and other things such as announcement of shapes is desired or annoying.

You can't ask users about everything and as I said, most defaults are sound.  

You could ask what features people most want added.  

Others may have other survey ideas.

**Comment by smeveriss** [comment]
Hi.
Well, I always change things round but admit because I am a portable user I almost never change things from fresh.
I do turn emphasis and a few other things off like mouse tracking by default but thats just me.
I see no issues.
One thing though excluding win7 and lower if nvda is loaded on modern windows thats 10 and up it should automatically install addonupdater and win10 apps as soon as it loads explaining it needs this, maybe giving a tutorial about addons, etc.
Either that or just do it as part of install or give the option during install to download core addons.
Arguably win10apps is needed for full functionality of the modern windows system and the addon updater is needed once any updated addons are installed.
If there is an addon store it needs to be installeed to for full use,etc.
Obviously you may have the situation where you may have to manually download whatever,.
Not sure how this would all but work though.
Its going to depend on the indevidual users what exactly is needed.
I accept when setting up windows that I will have to monkey round with all the settings.
I also tweak and hack settings to my liking within windows and my own tweaking and hacking modification prograsm.
Its a big list usually for me and there are a load for others, I couldn't possibly list all of them but most of those would be how I like my power plan set, themes, os interface, sounds, etc.
Really for nvda what is announceed right away and not the horrible extras.
Things llike words off would be a start and setting punctuation to well something where it will say get to the end and detect an end but not announce every full stop or capitalised letter.
However as I said I couldn't list every setting, I have my own prefference for whatever.
From a new user's prospective you probably need things to just work right away.

**Comment by Brian1Gaff** [comment]
I agree about Github, by the way, that site could also do with somebody who 
speaks in plain language to  make it less verbose.
 However, there was a survey right at the very start of NVDAs life and I 
think what has happened is that extra functionality has crept in slowly over 
time and unless defaults were considered bad by a lot of people they were 
indeed left as the  original  writer  decided. I always turn of block 
quotes, I turn a lot of stuff of in that document and browse stuff and 
change the settings of the mode switching, but that is purely me. I do have 
to use the  app settings  add on a lot there days to turn some of these back 
as they were to accommodate the use of an application.
 It would be really neat in some places to have a toggle but I do understand 
that keys are now getting scarce so to speak, and concert piano is not my 
Forte!

 As for a survey. It may well not be the appropriate place to force people 
to do it as a new screenreader user will be new to web forms, and possibly 
not know about many of the settings yet. It could however be somewhere as an 
option which would automatically record the system and nvda version 
versions and also include some usage questions. IE help us to help you 
survey.
 Brian

***@***.***
Sent via blueyonder.
Please address personal E-mail to:-
***@***.***, putting 'Brian Gaff'
in the display name field.
Newsgroup monitored: alt.comp.blind-users
----- Original Message ----- 
From: "Gene703122" ***@***.***>
To: "nvaccess/nvda" ***@***.***>
Cc: "Subscribed" ***@***.***>
Sent: Friday, March 04, 2022 1:53 AM
Subject: [nvaccess/nvda] A survey is needed to see if good defaults are 
being used (Discussion #13427)
